src/main/kotlin/no/elhub/auth/features/documents/create/PdfFactory.kt

```diff
@@ -0,0 +1,104 @@
+package no.elhub.auth.features.documents.create
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.core.left
+import arrow.core.raise.either
+import com.github.mustachejava.DefaultMustacheFactory
+import com.openhtmltopdf.pdfboxout.PdfRendererBuilder
+import org.apache.pdfbox.Loader
+import org.apache.pdfbox.pdmodel.PDDocumentInformation
+import java.io.ByteArrayOutputStream
+import java.io.StringWriter
+
+typealias MustacheVariableTags = Map<String, String>
+
+typealias PdfBytes = ByteArray
+
+sealed interface PdfError {
+    data object HtmlGenerationError : PdfError
+    data object PdfGenerationError : PdfError
+    data object MetaDataError : PdfError
+}
+
+fun interface PdfFactory {
+    fun create(command: CreateDocumentCommand): Either<PdfError, PdfBytes>
+}
+
+class HtmlToPdfFactory : PdfFactory {
+    private val mustacheFactory: DefaultMustacheFactory = DefaultMustacheFactory("templates")
+
+    enum class MustacheTemplateFile(val fileName: String) {
+        ChangeOfSupplier("change_of_supplier.mustache")
+    }
+
+    data class PdfData(
+        val template: MustacheTemplateFile,
+        val toBeSignedBy: String,
+        val tags: MustacheVariableTags
+    )
+
+    object PdfMetaData {
+        internal const val NIN = "NIN"
+    }
+
+    private fun CreateDocumentCommand.toPdfData(): PdfData = when (this) {
+        is CreateDocumentCommand.ChangeOfSupplier -> PdfData(
+            template = MustacheTemplateFile.ChangeOfSupplier,
+            toBeSignedBy = requestedFrom,
+            tags = mapOf(
+                "customerName" to requestedFromName,
```

Comment:
:art: Nice to use constants instead of concrete values; makes it easier to see things which could/should be extracted out to the configuration